src/Native/CMakeLists.txt

```diff
@@ -126,22 +126,20 @@ if(CLR_CMAKE_PLATFORM_ARCH_AMD64)
   add_definitions(-D_TARGET_AMD64_=1)
   add_definitions(-D_AMD64_)
   add_definitions(-DAMD64)
-  add_definitions(-DTARGET_X64)
-  add_definitions(-DTARGET_AMD64)
-  add_definitions(-D_WIN64=1)
+  if(WIN32)
+    add_definitions(-D_WIN64=1)
+  endif()
 elseif(CLR_CMAKE_PLATFORM_ARCH_I386)
   add_definitions(-D_TARGET_X86_=1)
   add_definitions(-D_X86_)
-  add_definitions(-DTARGET_X86)
-  add_definitions(-D_WIN32=1)
+  if(WIN32)
```

Comment:
_WIN32 should be always defined for Windows - both 32-bit and 64-bit.  It is documented [here](https://msdn.microsoft.com/en-us/library/b0084kay.aspx) .

It should be a new block - shared by all platforms. Something like:

```
if(WIN32)
     add_definitions(-D_WIN32=1)
     if(IS_64BIT_BUILD)
         add_definitions(-D_WIN64=1)
     endif(IS_64BIT_BUILD)
endif(WIN32)
```